src/dal_select2/static/autocomplete_light/select2.js

```diff
@@ -35,13 +35,20 @@
                     dstName = srcName;
                 }
                 // First look for this field in the inline
-                $field = $('[name=' + prefix + srcName + ']');
+                $field_selector = '[name=' + prefix + srcName + ']';
+                $field = $($field_selector);
                 if (!$field.length)
-                    // As a fallback, look for it outside the inline
-                    $field = $('[name=' + srcName + ']');
-                if ($field.length)
-                    forwardedData[dstName] = $field.val();
-
+                // As a fallback, look for it outside the inline
+                    $field_selector = '[name=' + srcName + ']';
+                $field = $($field_selector);
+                if ($field.length) {
+                    if ($field.attr('type') === 'checkbox')
+                        forwardedData[dstName] = !!$($field_selector + ":checked").val();
```

Comment:
I don't know if val() would return a string such as "unchecked" then the forwarded value would be `true` with the `!!` syntax. Did you consider this ?